driver/kubernetes/kubernetes.go

```diff
@@ -363,15 +370,37 @@ func (k *Driver) deleteJob(name string) error {
 }
 
 func generateNameTemplate(op *driver.Operation) string {
-	return fmt.Sprintf("%s-%s-", op.Installation, op.Action)
+	name := fmt.Sprintf("%s-%s", op.Action, op.Installation)
+	if len(name) > 50 {
+		name = name[0:39]
+	}
+
+	var result string
+	for _, match := range dns1123Reg.FindAllString(strings.ToLower(name), 10) {
```

Comment:
Why 10 occurrences?

Reply:
This was just an arbitrary number since I'm not aware of an API to allow unlimited matches (we're already constrained by the truncation above). Updated to use to max length instead.